pkg/metrics/metrics.go

```diff
@@ -0,0 +1,108 @@
+package metrics
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NodeEventFromNode returns a NodeEvent event from a node.
+func NodeEventFromNode(clusterID string, node corev1.Node) NodeEvent {
+	return NodeEvent{
+		ClusterID:   clusterID,
+		Labels:      node.Labels,
+		NodeName:    node.Name,
+		Info:        node.Status.NodeInfo,
+		Capacity:    node.Status.Capacity,
+		Allocatable: node.Status.Allocatable,
+		Role:        node.Labels["node.k0sproject.io/role"],
+	}
+}
+
+// NodeEvent is the event generated when a new node appears in the cluster
+// or an existing node is updated.
+type NodeEvent struct {
+	ClusterID   string                `json:"clusterID"`
+	Labels      map[string]string     `json:"labels"`
+	NodeName    string                `json:"nodeName"`
+	Info        corev1.NodeSystemInfo `json:"info"`
+	Capacity    corev1.ResourceList   `json:"capacity"`
+	Allocatable corev1.ResourceList   `json:"allocatable"`
```

Comment:
Including this in the hash is going to make you send a loooooot of updates I think?

Just want to make sure that's the intent

Reply:
every time a pod with a request is added/removed this will change

Reply:
Well, although expected this does not seem to happen. I have tested to create a deployment with requests and limits and that did not change the content of the allocatable field. Maybe that information has its own reconciliation ? I will wait a little bit longer and see if that reflects in the allocatable field.

Reply:
No change after 10 minutes. What I found out is that `allocatable = capacity - reservations`, reservations being thingies required to run system stuff such as kubelet.